core/src/main/java/fr/sncf/osrd/stdcm/graph/AllowanceManager.java

```diff
@@ -78,18 +80,30 @@ private STDCMEdge makeNewEdges(List<STDCMEdge> edges, Envelope totalEnvelope) {
                     .setPrevMaximumAddedDelay(maxAddedDelayAfter)
                     .setPrevAddedDelay(node == null ? 0 : node.totalPrevAddedDelay())
                     .setPrevNode(node)
-                    .setEnvelope(extractEnvelopeSection(totalEnvelope, toMeters(previousEnd), toMeters(end)))
+                    .setEnvelope(extractEnvelopeSection(totalEnvelope, previousEnd, end))
                     .setForceMaxDelay(true)
                     .setWaypointIndex(edge.waypointIndex())
                     .findEdgeSameNextOccupancy(edge.timeNextOccupancy());
             if (prevEdge == null)
                 return null;
             previousEnd = end;
         }
-        assert Math.abs(toMeters(previousEnd) - totalEnvelope.getEndPos()) < POSITION_EPSILON;
+        assert Math.abs(previousEnd - totalEnvelope.getEndPos()) < POSITION_EPSILON;
```

Comment:
maybe use arePositionsEqual?

Reply:
I'd add that the epsilon values in `TrainPhysicsIntegrator` should be passed from `public` to `private`, so as to enforce the use of the comparison methods you put in place :)

Reply:
Agree, but that would be a larger refactoring, for now I've only changed the places that were already edited in this PR